flutter/lib/src/default_integrations.dart

```diff
@@ -153,25 +153,17 @@ class NativeSdkIntegration extends Integration<SentryFlutterOptions> {
 /// See also:
 ///   - [SentryWidgetsBindingObserver]
 ///   - [WidgetsBindingObserver](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html)
-class WidgetsBindingIntegration extends Integration {
+class WidgetsBindingIntegration extends Integration<SentryFlutterOptions> {
   @override
-  FutureOr<void> call(Hub hub, SentryOptions options) {
-    if (options is SentryFlutterOptions) {
-      // We don't need to call `WidgetsFlutterBinding.ensureInitialized()`
-      // because `FlutterSentry.init` already calls it.
-      WidgetsBinding.instance.addObserver(SentryWidgetsBindingObserver(
-        hub: hub,
-        options: options,
-      ));
-
-      options.sdk.addIntegration('widgetsBindingIntegration');
-    } else {
-      options.logger(
-        SentryLevel.fatal,
-        'widgetsBindingIntegration failed to be installed because options is '
-        'not an instance of SentryFlutterOptions',
-      );
-    }
+  FutureOr<void> call(Hub hub, SentryFlutterOptions options) {
+    // We don't need to call `WidgetsFlutterBinding.ensureInitialized()`
+    // because `FlutterSentry.init` already calls it.
+    WidgetsBinding.instance.addObserver(SentryWidgetsBindingObserver(
+      hub: hub,
+      options: options,
+    ));
+
+    options.sdk.addIntegration('widgetsBindingIntegration');
```

Comment:
Does it make sense to return a bool from `Integration.call` indicating wether the integration was added
and have another method on `Integration` with the name? Then one would not have to add `options.sdk.addIntegration('widgetsBindingIntegration');` themself.
(Mybe try to see if `this.runtimeType.toString()` is an 
adequate name?)

Reply:
`this.runtimeType.toString()` is obfuscated at runtime if `split-debug` feature is enabled, so sadly not, it's not even possible to deobfuscated this yet, it's a limitation of the Dart tooling.